### ISSUE TITLE
Can't download videos from TikTok / Watermarks appear in videos downloaded from TikTok

### ISSUE THREAD

**Issue by MeyMan2007** [JunkFood02/Seal#1376]
### Checklist

- [ ] I'm reporting a bug unrelated to a specific site.
- [X] I've verified that I'm running the [**latest version**](https://github.com/yt-dlp/yt-dlp/releases/latest) of yt-dlp.
- [X] I've verified that I'm running the latest [**stable version**](https://github.com/JunkFood02/Seal/releases/latest/) of Seal or any later [**preview versions**](https://github.com/JunkFood02/Seal/releases).
- [X] I've read the [**Contributing guidelines**](https://github.com/JunkFood02/Seal/blob/main/CONTRIBUTING.md) and [**Code Of Conduct.**](https://github.com/JunkFood02/Seal/blob/main/CODE_OF_CONDUCT.md)
- [X] I've checked that the site i'm trying to download from is in the [**Supported Sites**](https://github.com/yt-dlp/yt-dlp/blob/master/supportedsites.md) list from yt-dlp
- [X] I understand that the issue will be (ignored/closed) if I intentionally remove or skip any mandatory field.

### To Reproduce

Try to download a video from TikTok by sharing it and using the quick download menu. 

### Error reports

App version: 1.12.0-beta.1 (11132)
Device information: Android 12 (API 31)
Supported ABIs: [arm64-v8a, armeabi-v7a, armeabi]
Yt-dlp version: 2024.03.10.232703

URL: 
WARNING: [TikTok] Expecting value in '': line 1 column 1 (char 0). Retrying... (attempt 1 of 4)
WARNING: [TikTok] Expecting value in '': line 1 column 1 (char 0). Retrying... (attempt 2 of 4)
WARNING: [TikTok] Expecting value in '': line 1 column 1 (char 0). Retrying... (attempt 3 of 4)
WARNING: [TikTok] 7254235270302797098: Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 1 column 1 (char 0)")); trying with webpage
ERROR: [TikTok] 7254235270302797098: Video not available, status code 0; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

**Comment by Ismail00008** [comment]
Have the same problem here . And when it does allow you to download a video it produces a watermark

**Comment by JunkFood02** [comment]
https://github.com/yt-dlp/yt-dlp/issues/9441

TL;DR

Settings - General - Update yt-dlp - switch to "Nightly" build and update it

**Comment by GhostBusterr4** [comment]
WARNING: [TikTok] Expecting value in '': line 1 column 1 (char 0). Retrying... (attempt 1 of 4)
WARNING: [TikTok] Expecting value in '': line 1 column 1 (char 0). Retrying... (attempt 2 of 4)
WARNING: [TikTok] Expecting value in '': line 1 column 1 (char 0). Retrying... (attempt 3 of 4)
WARNING: [TikTok] 7333335919656733957: Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 1 column 1 (char 0)")); trying with webpage
ERROR: unable to open for writing: [Errno 13] Permission denied:


How do I fix this problem?

**Comment by JunkFood02** [comment]
- https://github.com/yt-dlp/yt-dlp/issues/9506

**Comment by pukkandan** [comment]
@JunkFood02 You may wish to make yt-dlp-nightly the default for seal. It is our recommended build for most users now.

**Comment by JunkFood02** [comment]
> @JunkFood02 You may wish to make yt-dlp-nightly the default for seal. It is our recommended build for most users now.

Will do. Thank you for your(and the yt-dlp team's) hard work!

**Comment by VyrekXD** [comment]
Hey I'm still experiencing this issue, is this fixed on stable, preview or not resolved?

**Comment by VyrekXD** [comment]
Hey I'm still experiencing this issue, is this fixed on stable, preview or not resolved?

**Comment by JunkFood02** [comment]
@VyrekXD Update yt-dlp to nightly builds

**Comment by TonioGela** [comment]
> @VyrekXD Update yt-dlp to nightly builds

It doesn't seems that this fixes it. Is there a way to force the update of the underlying lib once you select nightly builds?

**Comment by TheBarbiter1** [comment]
> @VyrekXD Update yt-dlp to nightly builds

I'm on the nightly build, and I'm still getting this issue. And I actually like having the watermark saved (not the little jingle at the end). Most creators don't have a watermark in their videos, so leaving that in and being able to go back and find the original creator is really helpful

**Comment by zynpk007** [comment]
Watermark didn't remove after download please resolve this issue on priority

**Comment by JunkFood02** [comment]
Upstream: https://github.com/yt-dlp/yt-dlp/issues/9506
